chain/client/src/client.rs

```diff
@@ -804,9 +806,11 @@ impl Client {
 
                 match process_result {
                     ProcessPartialEncodedChunkResult::Known => Ok(vec![]),
-                    ProcessPartialEncodedChunkResult::HaveAllPartsAndReceipts(prev_block_hash) => {
-                        Ok(self
-                            .process_blocks_with_missing_chunks(prev_block_hash, protocol_version))
+                    ProcessPartialEncodedChunkResult::HaveAllPartsAndReceipts(_) => {
+                        self.chain
+                            .blocks_with_missing_chunks
+                            .accept_chunk(&chunk_hash, height_included);
```

Comment:
this doesn't work because `height_included` is not yet set at this point (it is 0).